openrndr-core/src/main/kotlin/org/openrndr/shape/ShapeContour.kt

```diff
@@ -574,70 +574,70 @@ data class ShapeContour(
         }
 
     /**
-     * Samples a new [ShapeContour] from the current [ShapeContour] starting at [startT] and ending at [endT].
+     * Samples a new [ShapeContour] from the current [ShapeContour] starting at [t0] and ending at [t1].
      *
-     * @param startT Starting point in range `0.0` to less than `1.0`.
-     * @param endT Ending point in range `0.0` to less than `1.0`.
+     * @param t0 Starting point in range `0.0` to less than `1.0`.
+     * @param t1 Ending point in range `0.0` to less than `1.0`.
      * @return Subcontour
      */
-    fun sub(startT: Double, endT: Double): ShapeContour {
+    fun sub(t0: Double, t1: Double): ShapeContour {
         if (empty) {
             return EMPTY
         }
 
-        require(startT == startT) { "u0 is NaN" }
-        require(endT == endT) { "u1 is NaN" }
+        require(!t0.isNaN()) { "u0 is NaN" }
```

Comment:
I strongly prefer the colloquial `x == x` over isNaN(). This also reads as a double negation.